ddclient

```diff
@@ -2164,10 +2165,22 @@ sub get_ip {
 	$arg = 'ip';
 
     } elsif ($use eq 'if') {
+        my $iface;
 	$skip  = opt('if-skip', $h)  || '';
-	$reply = `ifconfig $arg 2> /dev/null`;
-	$reply = `ip addr list dev $arg 2> /dev/null` if $?;
-	$reply = '' if $?;
+        if ($arg eq 'defaultroute') {
+            ($iface) = (grep(/^0\.0\.0\.0\b/, `route -n`))[0] =~ /(\S+)$/;
```

Comment:
A fallback to use `ip route ...` here would be useful (like all the other places a net-tools command is executed).

eg.
`ip -j -p route show default | grep '"dev":' | cut -d\" -f4`

Reply:
There are at least four ways to get the default route / interface that need to be handled.  I have an implementation in a fork that I have been working on.  I plan to submit it (not the entire fork, just parts of it) as a PR on ddclient once some of my other pre-req PRs are merged. https://github.com/dkerr64/ddclient/blob/a99d3333cb70bb6c77a97948ad57ca112eb0e1d8/ddclient.in#L2518

Using -j is an interesting thought, but unfortunately not available on all platforms / versions of iproute2.